### PR TITLE
Day5

### PR DESCRIPTION
**Pull request by dat-nnguyen**
completed Day 5, Day 6 is upcoming

### CODE REVIEW

File: Day5.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+#task 2
+new_list = ['I', 'fell', 'googley', 'coding', 'python']
+
+#task 3
+print(len(new_list))
+
+#task 4
+first_item, *middle_item, last_item = new_list
+print(first_item, middle_item, last_item)
```
Comment (tuang3142) [inline Day5.py:15]: incorrect. please check again. hint:
- `first_item = new_list[0]`
- the middle item should be 'googley'
- you can use a negative index to get the last item

File: Day5.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+
+#task 5
+mixed_data_types = ['Dat', 17, 185, 'single', 'Viet Nam']
```
Comment (tuang3142) [inline Day5.py:17]: wtf m85 rồi cơ à? proof?

File: Day5.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+#task 6, 7, 8, 9 
+it_companies = ['Facebook', 'Google', 'Microsoft', 'Apple', ' IBM', 'Oracle', 'Amazon']
+print(it_companies)
+print(len(it_companies))
+first_company, *middle_company, last_company = it_companies
+print(first_company, middle_company, last_company)
```
Comment (tuang3142) [inline Day5.py:26]: same as above: this is incorrect.

File: Day5.py [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+it_companies.append('Spotify')
+print(it_companies)
+
+#task 12
+it_companies.insert(3, 'Huawei')
+print(it_companies)
+
+#task 13
+it_companies[1].upper()
+print(it_companies)
```
Comment (tuang3142) [inline Day5.py:41]: `it_companies[1]` is still `Google`. It should be `GOOGLE`.

File: Day5.py [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+
+#task 14
+print(it_companies + ['#'])
```
Comment (tuang3142) [inline Day5.py]: incorrect. task 14 expects you to produce this string: `'Facebook#;Google#;Microsoft#;Apple#; IBM#;Oracle#;Amazon'`

hint: please research how to use python `join` method.

File: Day5.py [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+#task 15
+print('Facebook' in it_companies)
+
+#task 16 
+it_companies.sort()
+print(it_companies)
+
+#task 17
+it_companies.reverse()
+print(it_companies)
```
Comment (tuang3142) [inline Day5.py:56]: bonus: you can also use this syntax to reverse an array: `it_companies[::-1]` 😎

File: Day5.py [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+
+#task 18,19,20
+print(it_companies[:3])
+print(it_companies[-4:])
```
Comment (tuang3142) [inline Day5.py:60]: this should be it_companies[-3:]

File: Day5.py [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+print(it_companies[2:6:1])
```
Comment (tuang3142) [inline Day5.py:61]: this should be `it_companies[3:4]`. can you explain this syntax?

File: Day5.py [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+
+#task 25 
+it_companies.clear()
+print(it_companies)
+
+#task 26
+front_end = ['HTML', 'CSS', 'JS', 'React', 'Redux']
+back_end = ['Node','Express', 'MongoDB']
+joined_list = front_end + back_end
+print(joined_list)
```
Comment (tuang3142) [inline Day5.py:71]: asdfg

File: Day5.py [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+
+#task 27
+full_stack = joined_list.copy()
+full_stack.insert(6, 'Python')
+full_stack.insert(7, 'SQL')
+print(full_stack)
```
Comment (tuang3142) [inline Day5.py]: should be 5 and 6 instead of 6 and 7

File: Day5.py [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+
+
+#EXERCISE LEVEL 2
+#task 1
+ages = [19, 22, 19, 24, 20, 25, 26, 24, 25, 24]
+ages.sort()
+min_age = ages[0]
+print(min_age)
+max_age = ages[-1]
+print(max_age)
```
Comment (tuang3142) [inline Day5.py:91]: could also use `min(ages)` and `max(ages)`

File: Day5.py [hunk 12]
```diff
@@ -0,0 +1,313 @@
...
+
+ages.append(min_age)
+ages.append(max_age)
+print("Ages after adding min and max again:", ages)
+
+ages.sort()
+min_age = ages[0]
+max_age = ages[-1]
+#how to find the median age (one middle item or two middle items divided by two)
```
Comment (tuang3142) [inline Day5.py:100]: try this https://www.geeksforgeeks.org/find-median-of-list-in-python/

File: Day5.py [hunk 13]
```diff
@@ -0,0 +1,313 @@
...
+  'Vietnam',
+  'Yemen',
+  'Zambia',
+  'Zimbabwe',
+]
+first_country, *middle_country, last_country = countries 
+print(len(countries))
+
+divided_list = countries[0:96]
+print(divided_list)
```
Comment (tuang3142) [inline Day5.py]: a more correct solution would be:
```python
mid = len(countries) // 2
first_half, second_half = countries[:mid], countries[mid:]
```

File: Day6.py [hunk 14]
```diff
@@ -0,0 +1,56 @@
+#DAY 6 - TUPLES
+#Exercises: Level 1 
+
+#1. Create an empty tuple
+empty_tuple = ()
+
+#2. Create a tuple containing names of your sisters and your brothers (imaginary siblings are fine)
+my_bro = ('Tuan', 'Trung')
```
Comment (tuang3142) [inline Day6.py:8]: 'Chung' 🙃 But overall LGTM (looks good to me)

File: Day6.py [hunk 15]
```diff
@@ -0,0 +1,56 @@
+#DAY 6 - TUPLES
```
Comment (tuang3142) [inline Day6.py:1]: Gud to go 🍀

File: Day7.py [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+print(it_companies)
+#3. Insert multiple IT companies at once to the set it_companies
+others = {'Huawei', 'Xiaomi'}
+it_companies.union(others)
+print(it_companies)
+#4. Remove one of the companies from the set it_companies
+it_companies.remove('Facebook')
+print(it_companies) 
+#5. What is the difference between remove and discard?
+#Discard doesn't raise error, can be used without worrying about does the element exist in the set
```
Comment (tuang3142) [inline Day7.py:20]: good explaination 🧤

File: Day7.py [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+
+#6. What is the symmetric difference between A and B
+print(A.symmetric_difference(B))
+
+#7.Delete the sets completely
+del A 
+del B 
+
+#Exercises: Level 3
+age_set = [22, 19, 24, 25, 26, 24, 25, 24]
```
Comment (tuang3142) [inline Day7.py:51]: this should be `age_list`

File: Day7.py [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+
+#1. Convert the ages to a set and compare the length of the list and the set, which one is bigger?
+age_lst = set(age_set)
```
Comment (tuang3142) [inline Day7.py:54]: and this should be `age_set`

File: Day7.py [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+print(age_lst)
+print(len(age_set))
+print(len(age_lst))
```
Comment (tuang3142) [inline Day7.py:57]: why these two have different lengths?

File: Day7.py [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+
+#Exercises: Level 3
+age_set = [22, 19, 24, 25, 26, 24, 25, 24]
+
+#1. Convert the ages to a set and compare the length of the list and the set, which one is bigger?
+age_lst = set(age_set)
+print(age_lst)
+print(len(age_set))
+print(len(age_lst)) 
+#2. Explain the difference between the following data types: string, list, tuple and set
```
Comment (tuang3142) [inline Day7.py:58]: gud explaination 🐸

**Review by tuang3142 — COMMENTED** [review]
pls fix

File: Day5.py (hunk 2 above)
Comment (dat-nnguyen) [reply Day5.py:17]: nah, still 183cm TT

File: Day5.py (hunk 7 above)
Comment (dat-nnguyen) [reply Day5.py:60]: the first three companies bro

File: Day5.py (hunk 8 above)
Comment (dat-nnguyen) [reply Day5.py:61]: 18 is Slice out the first 3 companies from the list

19 is Slice out the last 3 companies from the list

20 Slice out the middle IT company or companies from the list

File: Day7.py (hunk 19 above)
Comment (dat-nnguyen) [reply Day7.py:57]: cuz set doesn't allow duplicate elements

**Pull request merged**